src/backend/distributed/shared_library_init.c

```diff
@@ -679,6 +679,20 @@ RegisterCitusConfigVariables(void)
 		GUC_UNIT_KB,
 		NULL, NULL, NULL);
 
+	DefineCustomBoolVariable(
+		"citus.sort_returning",
+		gettext_noop("Sorts the RETURNING clause to get consistent test output"),
+		gettext_noop("This feature is not intended for users. It is developed "
+					 "to get consistent regression test outputs. When enabled, "
+					 "the RETURNING clause returns the tuples sorted. The sort "
+					 "is done for all the entries, starting from the first one."
+					 "Finally, the sorting is done in ASC order."),
+		&SortReturning,
+		false,
+		PGC_SUSET,
```

Comment:
@marcoslot I'm curious what you think about `PGC_SUSET ` for this setting. Given that the code is not efficient, I'd suggest to keep it super-user only guc for not.

Reply:
> @marcoslot I'm curious what you think about PGC_SUSET for this setting. Given that the code is not efficient, I'd suggest to keep it super-user only guc for not.

I'm planning to merge this as-is for now. We can always revisit the decision.

Reply:
> I'm planning to merge this as-is for now. We can always revisit the decision.

Ah, sorry @pykello hasn't approved this yet 😄 So, I'm not merging now.

Reply:
Looks ok.

> > I'm planning to merge this as-is for now. We can always revisit the decision.
> 
> Ah, sorry @pykello hasn't approved this yet smile So, I'm not merging now.

Looks ok, with some minor comments. Doing another pass for more minor feedback.

Reply:
done with the review :smile:.